src/taipy/templates/multi-page-gui-core/{{cookiecutter.application_name}}/{{cookiecutter.application_main_file}}

```diff
@@ -0,0 +1,66 @@
+from taipy import Core
+from taipy.gui import Gui
+
+from pages.root.root import *
+from pages.scenario_page.scenario_page import scenario_page_md
+from pages.data_page.data_page import data_page_md
+
+from config.config import configure
+
+
+def on_init(state):
+    state.scenario = None
+    state.is_scenario_selected = False
+    state.is_scenario_computed = False
+    state.input_csv_file = None
+    state.replacement_type = "NO VALUE"
+    state.results = None
+
+
+def on_change(state, var: str, val):
+    if var == "scenario":
+        if val is not None:
+            state.is_scenario_selected = True
+        else:
+            state.is_scenario_selected = False
+    if var == "replacement_type":
+        state.scenario.replacement_type.write(val)
+    if var == "is_scenario_selected":
```

Comment:
All this code could go in the "if var == "scenario"" section

Reply:
I'm not sure to understand what you meant here.
If we put this block inside `if var == "scenario"`, the `var` can be both "scenario" and "replacement_type" at the same time.

Reply:
Same comment. I already simplified this locally.